### PR TITLE
Adding catwoman_tr to model transit limb-asymmetries

### PR DESCRIPTION
**Pull request by taylorbell57**
This PR makes progress on resolving #314 by adding support for the catwoman astrophysical model which can be used to model transit limb-asymmetry. catwoman is heavily based off of batman, so adding support for the model ended up being quite easy. No need to delay v1.0 for this PR, but just thought I'd publish it now since I'd got it working already

### CODE REVIEW

**Comment by taylorbell57** [comment]
Note that the pytest error is caused by an installation issue with catwoman which will be resolved by a [PR](https://github.com/KathrynJones1/catwoman/pull/17) that I've submitted to their repo

**Comment by taylorbell57** [comment]
@kevin218, a reminder that this PR is ready for you to review when you get the chance

File: src/eureka/S5_lightcurve_fitting/models/CatwomanModel.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        self.name = 'catwoman transit'
+        # Define transit model to be used
+        self.transit_model = partial(catwoman.TransitModel,
+                                     max_err=kwargs['max_err'],
+                                     fac=kwargs['fac'])
+
+        if ('rp2' not in self.longparamlist[0]
+                and 'rprs2' not in self.longparamlist[0]):
+            raise AssertionError('You must include an rp2 parameter in your '
+                                 'EPF when using catwoman.')
```
Comment (kevin218) [inline src/eureka/S5_lightcurve_fitting/models/CatwomanModel.py:34]: Don't we also need an `eval()` function?

**Review by kevin218 — CHANGES_REQUESTED** [review]

File: environment.yml [hunk 2]
```diff
@@ -36,6 +36,7 @@ dependencies:
     - tqdm
     - pip:
         - astraeus@git+https://github.com/kevin218/Astraeus@main
+        - catwoman@git+https://github.com/taylorbell57/catwoman@master
```
[the remark this answers is not in the document]
Comment (taylorbell57) [reply environment.yml:39]: Yeah, I have a PR on their repo that they haven't addressed

File: src/eureka/S5_lightcurve_fitting/models/CatwomanModel.py (hunk 1 above)
Comment (taylorbell57) [reply src/eureka/S5_lightcurve_fitting/models/CatwomanModel.py:34]: Nope, the catwoman class inherits that from the batman transit model in the same way the POET transit model does

**Review by kevin218 — APPROVED** [review]

**Pull request merged**